### PR TITLE
Serveral fixes for commands/townhall placeing and abadoning

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
- Fixed Townhall placement/abandon/delete
- Fixed/readded colony townhall teleport
- Added new setAbandon command
- Fixed LLama's attacking
- Fixed Mercenaries not opening doors

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/mobs/EntityMercenary.java [hunk 1]
```diff
@@ -128,7 +131,8 @@ public EntityMercenary(final EntityType<EntityMercenary> type, final World world
 
         this.goalSelector.addGoal(0, new SwimGoal(this));
         this.goalSelector.addGoal(1, new EntityMercenaryAI(this));
-        this.goalSelector.addGoal(2, new NearestAttackableTargetGoal<>(this, MonsterEntity.class, false));
+        this.goalSelector.addGoal(2, new OpenDoorGoal(this, true));
+        this.goalSelector.addGoal(3, new NearestAttackableTargetGoal(this, MonsterEntity.class, 10, true, false, e -> e instanceof IMob && !(e instanceof LlamaEntity)));
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/mobs/EntityMercenary.java]: Ahh nice

File: src/main/java/com/minecolonies/coremod/entity/mobs/EntityMercenary.java [hunk 2]
```diff
@@ -128,7 +131,8 @@ public EntityMercenary(final EntityType<EntityMercenary> type, final World world
 
         this.goalSelector.addGoal(0, new SwimGoal(this));
         this.goalSelector.addGoal(1, new EntityMercenaryAI(this));
-        this.goalSelector.addGoal(2, new NearestAttackableTargetGoal<>(this, MonsterEntity.class, false));
+        this.goalSelector.addGoal(2, new OpenDoorGoal(this, true));
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/mobs/EntityMercenary.java:135]: weren't they doing this anymore?

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/mobs/EntityMercenary.java:135]: not sure if it even works in 1.12 but in 1.14 they're stuck on doors

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/mobs/EntityMercenary.java:135]: You should probably also add the fence gate one (probably also for 1.12 then)

**Pull request merged**